File: app/src/main/java/com/lineclient/home/homelineclient/fragment/VideoFragment.java
package com.lineclient.home.homelineclient.fragment;

import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;

import com.lineclient.home.homelineclient.R;
import com.lineclient.home.homelineclient.net.HttpConnectHelper;
import com.lineclient.home.homelineclient.net.NetDataConstants;
import com.lineclient.home.homelineclient.net.ViewInterface;
import com.lineclient.home.homelineclient.tools.PlatformUtils;
import com.lineclient.home.homelineclient.view.PlayFrame;
import com.lineclient.home.homelineclient.view.RockerView;
import com.lineclient.home.homelineclient.ws.WsUtils;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.lang.ref.WeakReference;

/**
 * Created by yangfengyuan on 2017/7/25.
 */

public class VideoFragment extends BaseFragment implements View.OnClickListener {

    private View view;
    private WsUtils wsUtils;
    private PlatformUtils platformUtils;
    private RockerView.Direction directionState;
    private PlayFrame videoPlayerView;
    private Handler directionHandler = new Handler();
    private DirectionRunnable directionRunnable = new DirectionRunnable(this);
    private Button viewConnect;

    private static class DirectionRunnable implements Runnable {

        WeakReference<VideoFragment> weakReference;

        DirectionRunnable(VideoFragment object) {
            weakReference = new WeakReference<>(object);
        }

        @Override
        public void run() {
            VideoFragment object = weakReference.get();
            if (object != null) {
                if (object.directionState != null) {
                    object.platformUtils.handleRockerViewDirection(object.directionState);
                }
                object.directionHandler.postDelayed(object.directionRunnable, 100);
            }
        }
    }


    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {

        view = inflater.inflate(R.layout.fragment_video, container, false);
        wsUtils = WsUtils.getInstance(this.getContext());
        platformUtils = PlatformUtils.getInstance(wsUtils);

        initView();

        initData();

        return view;

    }

    private void initView() {

        initVideo();
        initControl();

    }

    private void initData() {

        wsUtils.addListener(new WsUtils.WSServiceInterface() {
            @Override
            public void serviceConnect() {

            }

            @Override
            public void serviceDisconnect() {

            }

            @Override
            public void onClose() {

            }

            @Override
            public void onError(String error) {

            }

            @Override
            public void onMessage(String msg) {

            }

            @Override
            public void onOpen() {

            }
        });
        wsUtils.startWsService();

        directionHandler.postDelayed(directionRunnable, 200);

    }

    void initVideo() {
        viewConnect = view.findViewById(R.id.video_connect);
        viewConnect.setOnClickListener(this);
        videoPlayerView = view.findViewById(R.id.player_root);
        videoPlayerView.init(this.getContext());

        videoPlayerView.setPath("rtmp://media3.sinovision.net:1935/live/livestream");
        try {
            videoPlayerView.start();
        } catch (IOException e) {
            // Toast.makeText(this,"播放失败",Toast.LENGTH_SHORT);
            e.printStackTrace();
        }


    }

    void initControl() {
        RockerView rockerView = view.findViewById(R.id.rockerview);
        if (rockerView != null) {
            rockerView.setCallBackMode(RockerView.CallBackMode.CALL_BACK_MODE_STATE_CHANGE);
            rockerView.setOnShakeListener(RockerView.DirectionMode.DIRECTION_8, new RockerView.OnShakeListener() {
                @Override
                public void onStart() {

                }

                @Override
                public void direction(RockerView.Direction direction) {
                    directionState = direction;
                }

                @Override
                public void onFinish() {
                    directionState = null;
                }
            });
        }
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.video_connect:
                viewConnect.setEnabled(false);
                getVideoUrl();
                break;
        }
    }

    public void getVideoUrl() {
        HttpConnectHelper.doAESPost((ViewInterface) this.getActivity(), NetDataConstants.GET_VIDEO_URL, null, null, new HttpConnectHelper.ResponseCallBack() {
            @Override
            public void callBack(String body) {
                try {
                    JSONObject jsonObject = new JSONObject(body);
                    if (jsonObject.getBoolean("success")) {
                        viewConnect.setVisibility(View.GONE);
                        String url = jsonObject.getString("url");
                        videoPlayerView.setPath(url);
                        try {
                            videoPlayerView.start();
                        } catch (IOException e) {
                           // Toast.makeText(this,"播放失败",Toast.LENGTH_SHORT);
                            e.printStackTrace();
                        }
                    } else {
                        viewConnect.setEnabled(true);
                    }
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        });
    }
}

File: app/src/main/java/com/lineclient/home/homelineclient/fragment/VideoFragment.kt
package com.lineclient.home.homelineclient.fragment

import android.os.Bundle
import android.os.Handler
import android.view.LayoutInflater
import android.view.View
import android.view.View.OnClickListener
import android.view.ViewGroup
import android.widget.Button
import com.lineclient.home.homelineclient.R
import com.lineclient.home.homelineclient.R.layout
import com.lineclient.home.homelineclient.net.HttpConnectHelper
import com.lineclient.home.homelineclient.net.NetDataConstants
import com.lineclient.home.homelineclient.net.ViewInterface
import com.lineclient.home.homelineclient.tools.PlatformUtils
import com.lineclient.home.homelineclient.view.PlayFrame
import com.lineclient.home.homelineclient.view.RockerView
import com.lineclient.home.homelineclient.view.RockerView.CallBackMode.CALL_BACK_MODE_STATE_CHANGE
import com.lineclient.home.homelineclient.view.RockerView.Direction
import com.lineclient.home.homelineclient.view.RockerView.DirectionMode.DIRECTION_8
import com.lineclient.home.homelineclient.view.RockerView.OnShakeListener
import com.lineclient.home.homelineclient.ws.WsUtils
import com.lineclient.home.homelineclient.ws.WsUtils.WSServiceInterface
import kotlinx.android.synthetic.main.fragment_video.player_root
import kotlinx.android.synthetic.main.fragment_video.video_connect
import org.json.JSONException
import org.json.JSONObject
import java.io.IOException
import java.lang.ref.WeakReference

/**
 * Created by yangfengyuan on 2017/7/25.
 */
class VideoFragment : BaseFragment(){
    private var views: View? = null
    private var wsUtils: WsUtils? = null
    private var platformUtils: PlatformUtils? = null
    private var directionState: Direction? = null
    //private var videoPlayerView: PlayFrame? = null
    private val directionHandler = Handler()
    private val directionRunnable = DirectionRunnable(this)
    private var viewConnect: Button? = null

    private class DirectionRunnable internal constructor(objects: VideoFragment) : Runnable {
        var weakReference: WeakReference<VideoFragment> = WeakReference(objects)
        override fun run() {
            val objects = weakReference.get()
            if (objects != null) {
                if (objects.directionState != null) {
                    objects.platformUtils!!.handleRockerViewDirection(objects.directionState)
                }
                objects.directionHandler.postDelayed(objects.directionRunnable, 100)
            }
        }

    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        views = inflater.inflate(layout.fragment_video, container, false)
        wsUtils = this.context?.let { WsUtils.getInstance(it) }
        platformUtils = PlatformUtils.getInstance(wsUtils)
        initView()
        initData()
        return views
    }

    private fun initView() {
        initVideo()
        initControl()
    }

    private fun initData() {
        wsUtils?.addListener(object : WSServiceInterface {
            override fun serviceConnect() {}
            override fun serviceDisconnect() {}
            override fun onClose() {}
            override fun onError(error: String?) {}
            override fun onMessage(msg: String?) {}
            override fun onOpen() {}
        })
        wsUtils!!.startWsService()
        directionHandler.postDelayed(directionRunnable, 200)
    }

    private fun initVideo() {
        video_connect.setOnClickListener{
            viewConnect!!.isEnabled = false
            videoUrl
        }
        player_root.init(this.context)
        player_root.setPath("rtmp://media3.sinovision.net:1935/live/livestream")
        try {
            player_root.start()
        } catch (e: IOException) {
            // Toast.makeText(this,"播放失败",Toast.LENGTH_SHORT);
            e.printStackTrace()
        }
    }

    fun initControl() {
        val rockerView: RockerView = view!!.findViewById(R.id.rockerview)
        if (rockerView != null) {
            rockerView.setCallBackMode(CALL_BACK_MODE_STATE_CHANGE)
            rockerView.setOnShakeListener(
                    DIRECTION_8, object : OnShakeListener {
                override fun onStart() {}
                override fun direction(direction: Direction) {
                    directionState = direction
                }

                override fun onFinish() {
                    directionState = null
                }
            })
        }
    }

    // Toast.makeText(this,"播放失败",Toast.LENGTH_SHORT);
    private val videoUrl: Unit
        get() {
            HttpConnectHelper.doAESPost(
                    (this.activity as ViewInterface?)!!, NetDataConstants.GET_VIDEO_URL, null, null
            ) { body ->
                try {
                    val jsonObject = JSONObject(body)
                    if (jsonObject.getBoolean("success")) {
                        viewConnect!!.visibility = View.GONE
                        val url = jsonObject.getString("url")
                        player_root.setPath(url)
                        try {
                            player_root.start()
                        } catch (e: IOException) {
                            // Toast.makeText(this,"播放失败",Toast.LENGTH_SHORT);
                            e.printStackTrace()
                        }
                    } else {
                        viewConnect!!.isEnabled = true
                    }
                } catch (e: JSONException) {
                    e.printStackTrace()
                }
            }
        }
}